### PR TITLE
Allow using PETSc solvers with CGL2 version 2

### PR DESCRIPTION
**Pull request by kmokstad**
Rebased #213 on top of #427

### CODE REVIEW

**Comment by kmokstad** [comment]
jenkins build this with downstreams please

**Comment by kmokstad** [comment]
jenkins build this with downstreams please

**Comment by kmokstad** [comment]
jenkins build this with downstreams please

**Comment by kmokstad** [comment]
jenkins build this with downstreams please

File: src/LinAlg/ProcessAdm.h [hunk 1]
```diff
@@ -60,7 +60,7 @@ class ProcessAdm
 #endif
 
   //! \brief The destructor releases the process administrator.
-  ~ProcessAdm();
+  virtual ~ProcessAdm();
```
Comment (kmokstad) [inline src/LinAlg/ProcessAdm.h:63]: I don't know why this came from the 2017 commit. As far as I know, ProcessAdm does not have sub-classes, so does the destructor needs to be virtual?

Comment (akva2) [reply src/LinAlg/ProcessAdm.h:63]: drop this hunk. slipped in back then from unrelated work is my guess. it's of no use.

**Comment by kmokstad** [comment]
jenkins build this with downstreams please

**Comment by kmokstad** [comment]
jenkins build this with downstreams please

**Pull request merged**